core/domain/services/assets/CoreAssetManager.php

```diff
@@ -0,0 +1,357 @@
+
+    const JS_HANDLE_EE_JS_API          = 'eejs-api';
+
+    const JS_HANDLE_EE_CORE            = 'espresso_core';
+
+    const JS_HANDLE_EE_I18N            = 'eei18n';
+
+    const JS_HANDLE_EE_ACCOUNTING      = 'ee-accounting';
+
+    const JS_HANDLE_EE_WP_PLUGINS_PAGE = 'ee-wp-plugins-page';
+
+    // EE CSS assets handles
+    const CSS_HANDLE_EE_DEFAULT = 'espresso_default';
+
+    const CSS_HANDLE_EE_CUSTOM  = 'espresso_custom_css';
+
+    /**
+     * @var EE_Currency_Config $currency_config
+     */
+    protected $currency_config;
+
+    /**
+     * @var EE_Template_Config $template_config
+     */
+    protected $template_config;
+
+
+    /**
+     * CoreAssetRegister constructor.
+     *
+     * @param AssetCollection    $assets
+     * @param EE_Currency_Config $currency_config
+     * @param EE_Template_Config $template_config
+     * @param DomainInterface    $domain
+     * @param Registry           $registry
+     */
+    public function __construct(
+        AssetCollection $assets,
+        EE_Currency_Config $currency_config,
+        EE_Template_Config $template_config,
+        DomainInterface $domain,
+        Registry $registry
+    ) {
+        $this->currency_config = $currency_config;
+        $this->template_config = $template_config;
+        parent::__construct($domain, $assets, $registry);
+    }
+
+
+    /**
+     * @since $VID:$
+     * @throws DuplicateCollectionIdentifierException
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidEntityException
+     */
+    public function addAssets()
+    {
+        $this->addManifestFile();
+        $this->addJavascriptFiles();
+        $this->addStylesheetFiles();
+    }
+
+
+    /**
+     * @since $VID:$
+     * @throws DuplicateCollectionIdentifierException
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidEntityException
+     */
+    public function addJavascriptFiles()
+    {
+        $this->loadCoreJs();
+        $this->loadJqueryValidate();
+        $this->loadAccountingJs();
+        add_action(
+            'AHEE__EventEspresso_core_services_assets_Registry__registerScripts__before_script',
+            array($this, 'loadQtipJs')
+        );
+        $this->registerAdminAssets();
+    }
+
+
+    /**
+     * @since $VID:$
+     * @throws DuplicateCollectionIdentifierException
+     * @throws InvalidDataTypeException
+     * @throws InvalidEntityException
+     */
+    public function addStylesheetFiles()
+    {
+        $this->loadCoreCss();
+    }
+
+
+    /**
+     * core default javascript
+     *
+     * @since $VID:$
+     * @throws DuplicateCollectionIdentifierException
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidEntityException
+     */
+    private function loadCoreJs()
+    {
+        $this->addJavascript(
+            CoreAssetManager::JS_HANDLE_EE_MANIFEST,
```

Comment:
I've always preferred using the class name for the following reasons:

 * it is explicit. You *always* know what class the constant (or property, or whatever) belongs to whereas with `self` you *need* to also know the class that the referenced code belongs to in order to know what `self` actually means. There have been times where people have posted code in chat (or forums, or whatever) that has included `self::something` but no mention of the class, which makes it impossible to know what that code is actually referring to. Your comment that "using self makes it clear the constant is from this class" has no value to me because why does it matter if the constant is from this class or another? How would anything change if the constant in the above code *was* from another class? IMO it doesn't change anything whatsoever. How does it change if we know that it *IS* from this class? Again, it doesn't. Knowing it is from *this* class has no benefit. Class constants work the same as static properties and are scoped globally, so it doesn't matter what class they come from as long as you know the actual class, but with `self` you can't tell that from looking at the constant alone.  I don't see how switching to using the explicit class name versus `self` would have any affect on how you would look at the code, because (using the code snippet above as an example) regardless of which is used, you would still need to look at the header for that code snippet (where it lists the file path `core/domain/services/assets/CoreAssetManager.php`) to see what class the code came from. In other words:
    - If the above code snippet had used `self::JS_HANDLE_EE_MANIFEST` then you would need to look at the snippet header to see the actual class that `self` referred to, because what is the point of knowing if a constant refers to *this* class if you don't know what *this* class is?
    - whereas when the above code snippet uses `CoreAssetManager::JS_HANDLE_EE_MANIFEST` then you only need to look at the snippet header to see that it belongs to *this* class. 
So either way you have to look at the snippet header to know where the code comes from. But again, knowing that a constant (or whatever) belongs to *this* class does not provide any extra benefit whatsoever, at least not in any way that I can see, but maybe you have some insight that I am completely overlooking.
- using the explicit class name instead of `self` makes it easier to copy n paste that constant (or whatever) into other classes that might need to reference it without having to do anything else but import the class. This will absolutely be the case if we switch over to using class constants for asset handles because those assets will be getting enqueued from some other class. So the other class will have no other option but to use  `CoreAssetManager::JS_HANDLE_EE_MANIFEST` so why not just use that everywhere? Although I don't think this is currently the case with any of our code, there is the potential to have a constant (or property, or whatever) with the same name in more than one class (imagine a static $ID property on multiple classes). In this (currently hypothetical) scenario, it should be obvious how using `self::$ID` could be disastrous or at least confusing, especially if you needed to reference that property from another class. Using the explicit class name avoids this issue completely.